File: Count_pairs_with_given_sum.py
def pairsCount(arr, n, sum):
    count = 0
    
    for i in range(0, n):
        for j in range(i+1, n):
            if arr[i] + arr[j] == sum:
                count += 1
    
    return count
    

arr = [1, 5, 7, -1, 5]
n = len(arr)
sum = 6
print("Count of pairs is : ", pairsCount(arr, n, sum))


# Output : Count of pairs is 3
# Source : https://www.geeksforgeeks.org/count-pairs-with-given-sum/

File: Count_pairs_with_given_sum.cpp
#include <bits/stdc++.h>

using namespace std;

int pairsCount(int arr[], int n, int sum)
{
    int count = 0;
    
    for(int i=0; i<n; i++)
    {
        for(int j=i+1; j<n; j++)
        {
            if(arr[i]+arr[j] == sum)
            {
                count++;
            }
        }
    }
    
    return count;
}

int main()
{
    int arr[] = {1, 5, 7, -1, 5};
    int n = sizeof(arr)/sizeof(arr[0]);
    int sum = 6;
    
    cout<<"Count of pairs is: "<<pairsCount(arr, n, sum);
}


// Output = Count of pairs is 3
// Source = https://www.geeksforgeeks.org/count-pairs-with-given-sum/
